pallets/rmrk-core/src/functions.rs

```diff
@@ -232,11 +249,10 @@ where
 		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
```

Comment:
What happens if we hit the `max_recursions`? Since this is called by `#[transactional]` will that cause the NFT to never get burned from storage since the error will rollback the Storage if this failed?

Reply:
let's say max_recursions is 5, and we have a 10-generation nft being burned.  my belief is that the first 5 generations will successfully burn, and subsequent descendants will fail, but need to test this.  opened #107 to address this.